app/routes/__public/oauth/$id.tsx

```diff
@@ -18,15 +18,40 @@ import randomString from "~/data/randomString.server";
 import { zOauthResponse } from "package/internal/types";
 export { default as ErrorBoundary } from "~/components/DefaultErrorBoundary";
 import { apiPost } from "package/internal/apiClient";
-import AES from "crypto-js/aes";
 
 const OauthConnectionPage = (): React.ReactElement => {
   const data = useLoaderData<typeof loadData>();
+  const [message, setMessage] = useState<string>("Please Wait");
 
   const postMessage = () => {
     if (window.opener && !window.opener.closed && "accessToken" in data) {
       window.opener.postMessage(data.accessToken, "*");
     }
+
+    if ("state" in data) {
+      let attemptAmount = 0;
+      const check = () => {
+        if (attemptAmount < 10) {
+          apiPost({
+            path: "access-token",
+            data: { state: data.state },
+          }).then((r) => {
+            if (r?.accessToken) {
+              setMessage("Success! This page will close.");
```

Comment:
"You may close this page" `window.close()` doesn't always close and I don't know when or why